web_decimal_numpad_dot/static/src/js/numpad_dot.js

```diff
@@ -19,14 +19,13 @@ odoo.define("web_decimal_numpad_dot.FieldFloat", function (require) {
                 ? ":" : translation._t.database.parameters.decimal_point;
         },
 
-        _replaceAt: function (cur_val, index, replacement) {
-            return cur_val.substr(0, index) + replacement +
-                   cur_val.substr(index + replacement.length);
+        _replaceAt: function (cur_val, from, to, replacement) {
+            return cur_val.substring(0, from) + replacement +
+                   cur_val.substring(to);
         },
 
         numpad_dot_replace: function (event) {
             // Only act on numpad dot key
-            event.stopPropagation();
```

Comment:
Because it didn't make sense in the first place.

Reply:
To explain a little better, if any handler reacts to a keypress, it would be inhibited as long as focus was in a float field. That didn't make sense, we don't care about event propagation in this case, the important part is to stop and modify the native behavior in case it's a numpad dot, and that's done below.